src/helper.cc

```diff
@@ -887,6 +891,26 @@ helper::handleFewerServers(const bool madeProgress)
     }
 }
 
+void
+helper::handleNoServers()
+{
+    Assure(!childs.n_active);
```

Comment:
We get here if GetFirstAvailable() returns false, which happens if childs.n_running is 0 or some other conditions are met (e.g., all other helpers have false flags.shutdown).

Reply:
I changed this assurance at 32d7a5f to avoid possible inconsistency between child.n_active and GetFirstAvailable() preconditions.

Reply:
Sounds reasonable.